src/software/ai/hl/stp/play/shoot_or_pass/shoot_or_pass_play_fsm.cpp

```diff
@@ -54,68 +64,81 @@ void ShootOrPassPlayFSM::lookForPass(const Update& event)
     if (event.common.num_tactics > 1)
     {
         ZoneNamedN(_tracy_look_for_pass, "ShootOrPassPlayFSM: Look for pass", true);
-        PassEvaluation<EighteenZoneId> pass_eval =
-            pass_generator.generatePassEvaluation(*event.common.world_ptr);
-        best_pass_and_score_so_far               = pass_eval.getBestPassOnField();
-        std::vector<EighteenZoneId> ranked_zones = pass_eval.rankZonesForReceiving(
-            *event.common.world_ptr, event.common.world_ptr->ball().position());
+        // Avoid passes to the goalie and the passing robot
+        std::vector<RobotId> robots_to_ignore = {};
+        auto friendly_goalie_id_opt =
+            event.common.world_ptr->friendlyTeam().getGoalieId();
+        if (friendly_goalie_id_opt.has_value())
+        {
+            robots_to_ignore.push_back(friendly_goalie_id_opt.value());
+        }
+        auto robot_with_ball_opt = event.common.world_ptr->friendlyTeam().getNearestRobot(
+            event.common.world_ptr->ball().position());
+        if (robot_with_ball_opt.has_value())
```

Comment:
Shouldn't this always have a value?

Reply:
our possession is decoupled from the `Play`s that are running. it's possible that there are a few ticks in which the ball is near an enemy robot but our AI hasn't decided to switch plays yet